src/commands/stop.ts

```diff
@@ -0,0 +1,57 @@
+import chalk from 'chalk';
+import BaseCommand from "../base-command";
+import { DockerComposeUtils } from "../common/docker-compose";
+import inquirer from 'inquirer';
+import { RequiresDocker } from '../common/docker/helper';
+
+export default class Stop extends BaseCommand {
+  async auth_required(): Promise<boolean> {
+    return false;
+  }
+
+  static description = 'Stop a local deployment';
+  static examples = [
+    'architect stop <local-environment-name>',
+  ];
+
+  static flags = {
+    ...BaseCommand.flags,
+  };
+
+  static args = [{
+    sensitive: false,
+    name: 'name',
+    description: 'Name of local environment',
+    required: false,
+  }];
+
+  @RequiresDocker({ compose: true })
+  async run(): Promise<void> {
+    const { args } = await this.parse(Stop);
+
+    const env_names = await DockerComposeUtils.getLocalEnvironments();
+    if (env_names.length === 0) {
+      throw new Error(chalk.red(`No local deployment found.`));
+    }
+
+    if (!args.name) {
+      const answers: { environment: string } = await inquirer.prompt([
```

Comment:
Neat trick that is used in other places.
```
default: flags.email,
when: !flags.email,
```
This will ask the user only if the flag is not set. If it is set then it won't ask and just set the answer to the flags value. This removes the conditional.

Reply:
That is a good trick.